File: framework/starter-logging/src/main/java/top/bettercode/logging/WebsocketProperties.java
package top.bettercode.logging;

import org.springframework.boot.context.properties.ConfigurationProperties;
import top.bettercode.lang.util.RandomUtil;

/**
 * RequestLogging 配置
 *
 * @author Peter Wu
 */
@ConfigurationProperties("summer.logging.websocket")
public class WebsocketProperties {

  /**
   * 是否启用
   */
  private boolean enabled = true;
  /**
   * 认证token
   */
  private String token = RandomUtil.nextString(16);
  //--------------------------------------------


  public boolean isEnabled() {
    return enabled;
  }

  public void setEnabled(boolean enabled) {
    this.enabled = enabled;
  }

  public String getToken() {
    return token;
  }

  public void setToken(String token) {
    this.token = token;
  }
}

File: web/src/main/kotlin/top/bettercode/summer/logging/WebsocketProperties.kt
package top.bettercode.summer.logging

import org.springframework.boot.context.properties.ConfigurationProperties
import top.bettercode.summer.tools.lang.util.RandomUtil.nextString

/**
 * RequestLogging 配置
 *
 * @author Peter Wu
 */
@ConfigurationProperties("summer.logging.websocket")
class WebsocketProperties {
    //--------------------------------------------
    /**
     * 是否启用
     */
    var isEnabled = true

    /**
     * 认证token
     */
    var token = nextString(16)
}